src/clique/CliqueManager.cc

```diff
@@ -266,12 +266,12 @@ void CliqueManager::SetByteLimits()
   m_allReduceByteLimit = rcclParamAllReduceCliqueByteLimit();
   if (m_allReduceByteLimit == 0)
   {
-    switch (m_gcnArch)
-    {
-    case 906: m_allReduceByteLimit =  16777216; break;
-    case 908: m_allReduceByteLimit =   8388608; break;
-    default:  m_allReduceByteLimit =  16777216; break;
-    }
+    if (strncmp(m_gcnArchName, "gfx906", 6) == 0)
```

Comment:
strncmp(archname, target, 6) gets repeated a lot through the code/

I think you can just add a helper function in archinfo,h

Perhaps
bool ArchMatches(char const* arch, char const* target)
{
  return strncmp(arch, target, strlen(target));
}

This should work even if you want to just compare for "gfx94"

Reply:
ah, that's a great idea!  Thank you.